### ISSUE TITLE
v2.12 regession: foo does not support item deletion (E1138)

### ISSUE THREAD

**Issue by skirpichev** [PyCQA/pylint#5407]
### Bug description

I got this error from [this line](https://github.com/diofant/diofant/blob/84c72e405aff6c87a40b6ff3281f51cc4f1fdb27/diofant/calculus/optimization.py#L240), where the `tableau` is a Matrix object, which has well defined `__delitem__()` method:
```
In [1]: m = Matrix([[1, 2], [3, 4]]); m
Out[1]: 
⎡1  2⎤
⎢    ⎥
⎣3  4⎦

In [2]: del m[-1, :]

In [3]: m
Out[3]: [1  2]

In [4]:                                       
```

See e.g. [this](https://github.com/diofant/diofant/runs/4339537793?check_suite_focus=true) build.

### Configuration

See configuration in the [pyproject.toml](https://github.com/diofant/diofant/blob/84c72e405aff6c87a40b6ff3281f51cc4f1fdb27/pyproject.toml) of the project.

### Command used

```shell
pylint diofant
```

### Pylint output

```shell
************* Module diofant.calculus.optimization
diofant/calculus/optimization.py:240:12: E1138: 'tableau' does not support item deletion (unsupported-delete-operation)

------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

No error.

### Pylint version

```shell
pylint-2.12.1
astroid-2.9.0
```

**Comment by Pierre-Sassoulas** [comment]
Hello, could you specify the version of numpy you're using ?

**Comment by skirpichev** [comment]
I doubt it does matter, but I can: from the quoted above build log - it's numpy-1.21.4.

**Comment by Pierre-Sassoulas** [comment]
The type of ``tableau`` and the version of the dependencie does matters as astroid sometime needs hints to parse code, so depending on the version the hint might be outdated or not.

**Comment by skirpichev** [comment]
@Pierre-Sassoulas, do you still think it's a false-positive?  I still able to reproduce this in the Github CI and the errors seems to be random.  Ex. in [this](https://github.com/diofant/diofant/actions/runs/1743832156) test run one job pass and another - fail.

**Comment by Pierre-Sassoulas** [comment]
I think it's a false positive as an error is reported even if the code works. If someone want to fix this it's going to be by modifying [numpy's brain in astroid](https://github.com/PyCQA/astroid/blob/main/astroid/brain/)

**Comment by skirpichev** [comment]
@Pierre-Sassoulas, it does matter, that the code doesn't use numpy at all?

**Comment by Pierre-Sassoulas** [comment]
Yes, it look like I'm confused about the type of Matrix. Could you show the import and provide the python code of your example please (you can edit the issue) ? Depending on what is in `Matrix` we need a new Brain or better astroid parsing capability.

**Comment by skirpichev** [comment]
> Could you show the import and provide the python code of your example please (you can edit the issue) ?

Sure, you can see imports on the top of the quoted in the original issue description file.  The Matrix class (it is the MutableDenseMatrix per default) is defined [here](https://github.com/diofant/diofant/blob/84c72e405aff6c87a40b6ff3281f51cc4f1fdb27/diofant/matrices/dense.py).  As you can see, it has the `__delitem__` method.

Unfortunately, I can't (yet) minimize the provided example.  E.g. if I run a targeted test `pylint diofant/calculus/optimization.py` - there is no error.

**Comment by skirpichev** [comment]
BTW, similar happens for [L237](https://github.com/diofant/diofant/blob/e623a88dc1ab6dfa1d1be60b189f980b844db6e7/diofant/calculus/optimization.py#L237) if I enable the 'unsubscriptable-object' warning.  @Pierre-Sassoulas, let me know if it's better to fill a separate issue.

**Comment by skirpichev** [comment]
I think, this is fixed in 2.13+ (my recent [pr](https://github.com/diofant/diofant/pull/1225) removed the pylint pragma).
